src/test/java/com/dell/isg/smi/service/server/configuration/manager/ConfigurationManagerImplTest.java

```diff
@@ -0,0 +1,219 @@
+            attr = new Attribute();
+            attr.setName("RAIDHotSpareStatus");
+            attr.setValue("Yes");
+            nestedComponent.getAttributes().add(attr);
+            subComponent.getNestedComponents().add(nestedComponent);
+            component.getSubComponents().add(subComponent);
+
+
+            request.setServerComponents(serverComponents);
+
+            configManager.updateComponents(request);
+
+            DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+            DocumentBuilder db = dbf.newDocumentBuilder();
+
+            Document document = db.parse(tempFile.toFile());
+            JAXBContext jaxbContext = JAXBContextFactory.createContext(new Class[] { SystemConfiguration.class },
+                                                                       null);
+
+            Binder<Node> binder = jaxbContext.createBinder();
+            binder.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+            binder.setProperty(Marshaller.JAXB_FRAGMENT, true);
+
+            SystemConfiguration systemConfig = (SystemConfiguration) binder.unmarshal(document);
+            assertNotNull(systemConfig);
+            for (ServerComponent serverComponent : systemConfig.getServerComponents()) {
+                if (serverComponent.getFQDD().equals("BIOS.Setup.1-1")) {
+                    for (Attribute attribute : serverComponent.getAttributes()) {
+                        switch (attribute.getName()) {
+                            case "MemTest":
+                                assertTrue(attribute.getValue().equals("Enabled"));
+                                break;
+                            case "SecureBoot":
+                                assertTrue(attribute.getValue().equals("Enabled"));
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+                } else if (serverComponent.getFQDD().equals("RAID.Slot.1-1")) {
+                    for (SubComponent sub : serverComponent.getSubComponents()) {
+                        for (NestedComponent nested : subComponent.getNestedComponents()) {
+                            if (nested.getFQDD(). equals("Disk.Bay.0:Enclosure.Internal.0-0:RAID.Slot.1-1")) {
+                                for (Attribute attribute : nestedComponent.getAttributes()) {
+                                    if (attribute.getName().equals("RAIDHotSpareStatus")) {
+                                        assertTrue(attribute.getValue().equals("Yes"));
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+        } finally {
+            Files.deleteIfExists(tempFile);
+        }
+    }
+
+    Node textToNode(DocumentBuilder documentBuilder, String text) throws IOException, SAXException {
```

Comment:
Looks like everything here and below is just exploratory, `configManager` is not being called. Just delete since it's not actually testing code.